vassal-app/src/main/java/VASSAL/build/module/turn/TurnComponent.java

```diff
@@ -26,13 +26,22 @@
 import VASSAL.build.BadDataReport;
 import VASSAL.build.Buildable;
 import VASSAL.build.module.documentation.HelpFile;
+import VASSAL.configure.ComponentDescription;
 import VASSAL.i18n.Resources;
 import VASSAL.tools.ErrorDialog;
 
 /**
  * Generic Turn Component
  */
-public class TurnComponent extends AbstractConfigurable {
+public class TurnComponent extends AbstractConfigurable implements ComponentDescription {
+  public static final String DESCRIPTION = "description"; //NON-NLS
+
+  protected String description;
+
+  @Override
+  public String getDescription() {
+    return description;
+  }
```

Comment:
What sets it?

Reply:
It gets set by the setAttribute()

Reply:
(and I should explain that TurnComponent, even though it isn't declared abstract, is ONLY overridden by the various fleshed-out kinds of turn component)

Reply:
The horror, the horror. Ok, np.